hc/api/management/commands/sendreports.py

```diff
@@ -28,27 +28,91 @@ def add_arguments(self, parser):
         )
 
     def handle_one_run(self):
+        """
+        Method to send emails for daily, weekly or monthly
+        based on user preference. Method is background worker.
+        """
+        # get current time
         now = timezone.now()
-        month_before = now - timedelta(days=30)
+        # get time stamp for day before to
+        # ensure user only gets email at the earliest at
+        # least 1 day after subscribing
+        day_before = now - timedelta(days=1)
```

Comment:
***CodeQuality: Critical***
There is a lot of repetition going on here. There should be a function that abstracts this out and used for daily weekly and monthly